docs/index.asciidoc

```diff
@@ -128,50 +159,88 @@ input {
 
 [id="plugins-{type}s-{plugin}-oid_root_skip"]
 ===== `oid_root_skip` 
 
-ToDO: Update description and values with correct info
+The `oid_root_skip` option specifies the  number of OID root digits to ignore in event field name.
+For example, in a numeric OID like "1.3.6.1.2.1.1.1.0" the first 5 digits could be ignored by setting `oid_root_skip => 5`
+which would result in a field name "1.1.1.0". Similarly when a MIB is used an OID such
+"1.3.6.1.2.mib-2.system.sysDescr.0" would become "mib-2.system.sysDescr.0"
 
-* Value type is <<boolean,boolean>>
-* Default value is `0` (`false`)
+* Value type is <<number,number>>
+* Default value is `0`
 
-[id="plugins-{type}s-{plugin}-walk"]
-===== `walk` 
+==== SNMPv3 Authentication Options
 
-ToDO: Update description and values with correct info
+A **single user** can be configured and will be used for all defined SNMPv3 hosts.
+Multiple snmp input declarations will be needed if multiple SNMPv3 users are required.
+These options are not required when not using SNMPv3.
 
-* Value type is <<array,array>>
+[id="plugins-{type}s-{plugin}-security_name"]
+===== `security_name`
+
+The `security_name` option specifies the SNMPv3 security name or user name
+
+* Value type is <<string,string>>
 * There is no default value for this setting
 
-Queries the subtree of information starting at the given OID(s).
 
-Example
-[source,ruby]
------
-  snmp {
-    walk => ["1.3.6.1.2.1.1"]
-    hosts => [{host => "udp:127.0.0.1/161" community => "public"}]
-  }
-}
------
+[id="plugins-{type}s-{plugin}-auth_protocol"]
+===== `auth_protocol`
+
+The `auth_protocol` option specifies the SNMPv3 authentication protocol or type
+
+* Value can be any of: `md5`, `sha`, `sha2`, `hmac128sha224`, `hmac192sha256`, `hmac256sha384`, `hmac384sha512`
+* There is no default value for this setting
+
+[id="plugins-{type}s-{plugin}-auth_pass"]
+===== `auth_pass`
+
+The `auth_pass` option specifies the SNMPv3 authentication passphrase or password
+
+* Value type is <<password,password>>
+* There is no default value for this setting
+
+[id="plugins-{type}s-{plugin}-priv_protocol"]
+===== `priv_protocol`
+
+The `priv_protocol` option specifies the SNMPv3 privacy/encryption protocol
+
+* Value can be any of: `des`, `3des`, `aes`, `aes128`, `aes192`, `aes256`
+* There is no default value for this setting
+
+[id="plugins-{type}s-{plugin}-priv_pass"]
+===== `priv_pass`
+
+The `priv_pass` option specifies the SNMPv3 encryption password
+
+* Value type is <<password,password>>
+* There is no default value for this setting
+
+[id="plugins-{type}s-{plugin}-security_level"]
+===== `security_level`
+
+The `security_level` option specifies the SNMPv3 security level between Authentication, No Privacy; Authentication, Privacy; or no Authentication, no Privacy
+
+* Value can be any of: `noAuthNoPriv`, `authNoPriv`, `authPriv`
+* There is no default value for this setting
 
 ==== More configuration examples
```

Comment:
Since the options in the config examples are for `SNMP Input Configuration Options`, maybe we shoud move the examples up into that section.